src/libpmemobj/ulog.h

```diff
@@ -76,13 +76,22 @@ struct ulog_entry_buf {
 	uint64_t next; /* offset of ulog extension */\
 	uint64_t capacity; /* capacity of this ulog in bytes */\
 	uint64_t gen_num; /* generation counter */\
-	uint64_t unused[4]; /* must be 0 */\
+	uint64_t flags; /* ulog flags */\
+	uint64_t unused[3]; /* must be 0 */\
 	uint8_t data[capacity_bytes]; /* N bytes of data */\
 }\
 
 #define SIZEOF_ULOG(base_capacity)\
 (sizeof(struct ulog) + base_capacity)
 
+/*
+ * Ulog preallocated by the user must be marked by this flag.
+ * It is important to not free it at the end:
+ * what user has allocated - user should free himself.
+ */
+#define ULOG_PREALLOCATED (1U << 0)
+#define ULOG_CURRENTLY_USED (1U << 1)
```

Comment:
this is debug only, right? Please #ifdef all of it.

Reply:
Ok, nvm, I get it now.
Would it make sense to remove the ifdefs and do this as a simple `if` so that we can switch this at runtime?